### ISSUE TITLE
Suggestion: Add prefix to enum types

### ISSUE THREAD

**Issue by acetinick** [DmitryTsepelev/store_model#84]
Hi,  

Great library, have migrated already out settings stores to this library and works great.  

One thing that it would be great is for enum types to support prefix settings.

Currently to avoid conflicts for many enum settings it would be good to have this similar feature from enum rails core

eg.
```
enum :authentication_method, %i[email phone], default: :email
enum :notification_method, %i[email phone], default: :email
```

currently we get arround this by doing this,

```
enum :authentication_method, %i[authentication_method_email authentication_method_phone], default: :authentication_method_email
enum :notification_method, %i[notification_method_email notification_method_phone], default: :notification_method_email
```
however, would be cleaner to have

```
enum :authentication_method, %i[email phone], default: :email, _prefix: true
enum :notification_method, %i[email phone], default: :email, _prefix: true

```

**Comment by DmitryTsepelev** [comment]
Hi @acetinick, nice to hear that the gem was useful to you 🙂 This feature definitely makes sense, do you want to take a lead on implementing it?

**Comment by acetinick** [comment]
Yep sure, would be something I can help out with over the coming weeks.

**Comment by DmitryTsepelev** [comment]
Implemented in #102
